### ISSUE TITLE
Misleading structure "matches"

### ISSUE THREAD

**Issue by happy-lorna** [ProteinsWebTeam/interpro7-client#73]
We should only report structure matches where the sequence location of the structure overlaps with the location of the signature/entry match. Otherwise we are reporting structures for various other domains which happen to be also found on that protein, but which are not relevant to the entry/signature in question.
e.g. https://www.ebi.ac.uk/interpro/beta/entry/InterPro/IPR001202/structure/PDB/#table
and
https://www.ebi.ac.uk/interpro/beta/entry/pfam/PF18507/structure/PDB/#table

**Comment by gustavo-salazar** [comment]
I am aware of this issue, but potentially requires a lot of development in the release procedures.
The situation extends to other parts of the website: for example: https://www.ebi.ac.uk/interpro/beta/structure/PDB/2c5z/

Not sure how big would be the impact of these in other parts, for instance counters?
What do you think @matthiasblum ?

The good thing is that you can see that they are not matching, which speaks well of the visualisation! 😛

**Comment by matthiasblum** [comment]
For counters, it's rather straightforward. We simply need to check whether the chain of a PDBe structure overlap with at least one fragment of an entry match.

But for Elasticsearch is a bit trickier. As we store _relationships_ (i.e. all possible combinations between proteins, proteomes, structures, chains, entries, and sets) additional checks have to be implemented when evaluating whether an entry and a structure overlap.

For instance, let assume we have a protein `ROLF_PANDA` whose 3D structure `1iv3` has been determined (this structure has two chains: `A`, and `B`). `ROLF_PANDA` is matched by two entries: `BATMAN`, and `ROBIN`. If we ignore proteomes and sets, we'd have to following Elastic documents:

- `ROLF_PANDA`-`BATMAN`-`1iv3`-`A`
- `ROLF_PANDA`-`BATMAN`-`1iv3`-`B`
- `ROLF_PANDA`-`ROBIN`-`1iv3`-`A`
- `ROLF_PANDA`-`ROBIN`-`1iv3`-`B`

However, it turns out that `BATMAN` does not overlap neither `A` nor `B`, and that `ROBIN` only overlaps `A`. Three out the four documents are now obsolete

- ~`ROLF_PANDA`-`BATMAN`-`1iv3`-`A`~
- ~`ROLF_PANDA`-`BATMAN`-`1iv3`-`B`~
- `ROLF_PANDA`-`ROBIN`-`1iv3`-`A`
- ~`ROLF_PANDA`-`ROBIN`-`1iv3`-`B`~

But now there are several missing relationships:

- Protein `ROLF_PANDA` should be associated with chain `B`
- Protein `ROLF_PANDA` should be associated with entry `BATMAN`

So we have to create the following documents:

- `ROLF_PANDA`-`1iv3`-`B`
- `ROLF_PANDA`-`BATMAN`

---

@gustavo-salazar would this impact the way the API is working?

**Comment by gustavo-salazar** [comment]
I don't think this would affect the API as it always aggregates looking for unique values. But I would have to properly test this with the fixtures data first, just to make sure I'm not missing anything. 
We know that batman is already lost, but let's make sure we keep robin safe!
